src/shogun/machine/Composite.h

```diff
@@ -56,7 +56,12 @@ namespace shogun
 			m_stages = std::forward<T>(stages);
 		}
 
-		std::shared_ptr<EnsembleMachine> train(
+#ifdef SWIG
+		std::shared_ptr<Machine>
```

Comment:
Shouldn’t it always return Machine?

Reply:
I would prefer that, or possibly return nothing at all, and add another function to return the internal machine since it's mostly not needed.

Reply:
what do you suggest?

Reply:
I like the returning of the fitted object from train. But it should be of type Machine consistently. This is what scikit-learn does, and it works well imo